iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/NettyTNonblockingTransport.java

```diff
@@ -0,0 +1,621 @@
+    }
+    if (connecting.get()) {
+      try {
+        TimeUnit.MILLISECONDS.sleep(1);
+      } catch (InterruptedException e) {
+        throw new IOException(e);
+      }
+    }
+    synchronized (lock) {
+      boolean dummyFinished = dummyClient.finishConnect();
+      boolean isConnected = connected.get() && dummyFinished;
+      if (logger.isDebugEnabled()) {
+        logger.debug(
+            "finishConnect called, netty connected: {}, dummy finished: {}",
+            connected.get(),
+            dummyFinished);
+      }
+      return isConnected;
+    }
+  }
+
+  @Override
+  public SelectionKey registerSelector(Selector selector, int interests) throws IOException {
+    synchronized (lock) {
+      this.selector = selector;
+      return dummyClient.register(selector, interests);
+    }
+  }
+
+  @Override
+  public String toString() {
+    synchronized (lock) {
+      return "[remote: " + getRemoteAddress() + ", local: " + getLocalAddress() + "]";
+    }
+  }
+
+  public SocketAddress getRemoteAddress() {
+    synchronized (lock) {
+      return channel != null ? channel.remoteAddress() : null;
+    }
+  }
+
+  public SocketAddress getLocalAddress() {
+    synchronized (lock) {
+      return channel != null ? channel.localAddress() : null;
+    }
+  }
+
+  private class NettyTransportHandler extends ChannelInboundHandlerAdapter {
+
+    @Override
+    public void channelActive(ChannelHandlerContext ctx) throws Exception {
+      if (logger.isDebugEnabled()) {
+        logger.debug("Channel active: {}", ctx.channel().remoteAddress());
+      }
+      super.channelActive(ctx);
+    }
+
+    @Override
+    public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+      if (msg instanceof ByteBuf) {
+        ByteBuf byteBuf = (ByteBuf) msg;
+        if (logger.isDebugEnabled()) {
+          logger.debug("Received {} bytes", byteBuf.readableBytes());
+        }
+        try {
+          synchronized (lock) {
+            readQueue.offer(byteBuf.retain());
+            // Trigger OP_READ on dummy by writing dummy byte
+            if (dummyServerAccepted != null) {
+              ByteBuffer dummyByte = ByteBuffer.wrap(new byte[1]);
+              dummyServerAccepted.write(dummyByte);
+            }
+            // Wakeup selector if needed
+            if (selector != null) {
+              selector.wakeup();
+            }
+          }
+        } finally {
+          byteBuf.release();
+        }
+      }
+    }
+
+    @Override
+    public void channelInactive(ChannelHandlerContext ctx) throws Exception {
+      if (logger.isDebugEnabled()) {
+        logger.debug("Channel inactive: {}", ctx.channel().remoteAddress());
+      }
+      synchronized (lock) {
+        if (dummyServerAccepted != null) {
+          dummyServerAccepted.close();
+          dummyServerAccepted = null;
+        }
+        if (channel != null) {
+          channel.close();
+          channel = null;
+        }
```

Comment:
How about dummyServer and dummyClient. 
Or `NettyTNonblockingTransport.this.close`?

Reply:
`NettyTNonblockingTransport.close` will be called in `TAsyncClient.onError` method. So I think it is OK to keep the current implement?